libsolidity/ast/AST.h

```diff
@@ -1174,6 +1174,76 @@ class IfStatement: public Statement
 	ASTPointer<Statement> m_falseBody; ///< "else" part, optional
 };
 
+/**
+ * Clause of a try-catch block. Includes both the successful case and the
+ * insuccessful cases.
+ * Names are only allowed for the insuccessful cases.
+ */
+class TryCatchClause: public ASTNode, public Scopable
+{
+public:
+	TryCatchClause(
+		SourceLocation const& _location,
+		ASTPointer<ASTString> const& _errorName,
+		ASTPointer<ParameterList> const& _parameters,
+		ASTPointer<Block> const& _block
+	):
+		ASTNode(_location),
+		m_errorName(_errorName),
+		m_parameters(_parameters),
+		m_block(_block)
+	{}
+	void accept(ASTVisitor& _visitor) override;
+	void accept(ASTConstVisitor& _visitor) const override;
+
+	ASTString const& errorName() const { return *m_errorName; }
+	ParameterList const* parameters() const { return m_parameters.get(); }
+	Block const& block() const { return *m_block; }
+
+private:
+	ASTPointer<ASTString> m_errorName;
+	ASTPointer<ParameterList> m_parameters;
+	ASTPointer<Block> m_block;
+};
+
+/**
+ * Try-statement with a variable number of catch statements.
+ * Syntax:
+ * try <call> returns (uint x, uint y) {
+ *   // success code
+ * } catch Error(string memory cause) {
+ *   // error code, reason provided
+ * } catch (bytes memory lowLevelData) {
```

Comment:
Does the order of catch clauses matter? All the examples follow the rule `catch Error` < `low level catch` if both are present, but I haven't seen that being enforced in the code.

Reply:
No, it does not.

Reply:
Ah ok